resources/docs/TUTORIAL_8_MODEL_OPTIMIZATION.md

```diff
@@ -0,0 +1,156 @@
+Additionally, you can specify the number of runs per evaluation run (`training_runs`). 
+If you specify more than one training run, one evaluation run will be executed the specified number of times.
+The final evaluation score will be the average over all those runs.
+
+```python
+from flair.hyperparameter.param_selection import TextClassifierParamSelector, OptimizationValue
+
+# what label do we want to predict?
+label_type = 'question_class'
+
+# create the parameter selector
+param_selector = TextClassifierParamSelector(
+    corpus,
+    label_type, 
+    False, 
+    'resources/results', 
+    'lstm',
+    max_epochs=50, 
+    training_runs=3,
+    optimization_value=OptimizationValue.DEV_SCORE
+)
+
+# start the optimization
+param_selector.optimize(search_space, max_evals=100)
+```
+
+The parameter settings and the evaluation scores will be written to `param_selection.txt` in the result directory.
+While selecting the best parameter combination we do not store any model to disk. We also do not perform a test run
+during training, we just evaluate the model once after training on the test set for logging purpose.
+
+## Finding the best Learning Rate
+
+The learning rate is one of the most important hyper parameter and it fundamentally depends on the topology of the loss
+landscape via the architecture of your model and the training data it consumes. An optimal learning will improve your
+training speed and hopefully give more performant models. A simple technique described by Leslie Smith's
+[Cyclical Learning Rates for Training](https://arxiv.org/abs/1506.01186) paper is to train your model starting with a
+very low learning rate and increases the learning rate exponentially at every batch update of SGD. By plotting the loss
+with respect to the learning rate we will typically observe three distinct phases: for low learning rates the loss does
+not improve, an optimal learning rate range where the loss drops the steepest and the final phase where the loss
+explodes as the learning rate becomes too big. With such a plot, the optimal learning rate selection is as easy as
+picking the highest one from the optimal phase.
+
+In order to run such an experiment start with your initialized `ModelTrainer` and call `find_learning_rate()` with the
+`base_path` and the optimizer (in our case `torch.optim.adam.Adam`). Then plot the generated results via the
+`Plotter`'s `plot_learning_rate()` function and have a look at the `learning_rate.png` image to select the optimal
+learning rate:
+
+```python
+from flair.datasets import WNUT_17
+from flair.embeddings import TokenEmbeddings, WordEmbeddings, StackedEmbeddings
+from flair.trainers import ModelTrainer
+from typing import List
+from torch.optim.adam import Adam
+
+# 1. get the corpus
+corpus = WNUT_17().downsample(0.1)
+print(corpus)
+
+# 2. what tag do we want to predict?
+tag_type = 'ner'
+
+# 3. make the tag dictionary from the corpus
+tag_dictionary = corpus.make_tag_dictionary(tag_type=tag_type)
```

Comment:
should be `tag_dictionary = corpus.make_label_dictionary(label_type=tag_type)`, otherwise the example won't run